jaxrs2/src/main/java/io/opentracing/contrib/jaxrs/server/OperationNameProvider.java

```diff
@@ -0,0 +1,56 @@
+package io.opentracing.contrib.jaxrs.server;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * @author Pavol Loffay
+ */
+public interface OperationNameProvider {
+
+    String operationName(ContainerRequestContext containerRequestContext);
+
+    OperationNameProvider HTTP_METHOD_NAME_PROVIDER = new OperationNameProvider() {
+        @Override
+        public String operationName(ContainerRequestContext containerRequestContext) {
+            return containerRequestContext.getMethod();
+        }
+    };
+
+    OperationNameProvider HTTP_PATH_NAME_PROVIDER = new OperationNameProvider() {
+        @Override
+        public String operationName(ContainerRequestContext containerRequestContext) {
+            String path = containerRequestContext.getUriInfo().getPath();
+            if (path != null && path.startsWith("/")) {
+                path = path.substring(1);
+            }
+
+            return path;
+        }
+    };
+
+    OperationNameProvider HTTP_PATH_WILDCARD_NAME_PROVIDER = new OperationNameProvider() {
+        @Override
+        public String operationName(ContainerRequestContext containerRequestContext) {
+            MultivaluedMap<String, String> pathParameters = containerRequestContext.getUriInfo().getPathParameters();
+
+            String path = containerRequestContext.getUriInfo().getPath();
+            if (path != null && path.startsWith("/")) {
+                path = path.substring(1);
+            }
+
+            for (Map.Entry<String, List<String>> entry: pathParameters.entrySet()) {
```

Comment:
Please, leave a javadoc for the expected behavior with examples. For instance, I'd like to know what's the expected behavior for the URL `/foo/bar/123`, which reaches a method annotated with `@Path("/foo/bar/{id : \\d+}")`, as well as the URL `/foo/bar/name`, which reaches a method annotated with `@Path("/foo/bar/{name: \\w+}")`